frontend/src/scenes/retention/RetentionLineGraph.tsx

```diff
@@ -43,17 +43,17 @@ export function RetentionLineGraph({
         <>
             <LineGraph
                 data-attr="trend-line-graph"
-                type="line"
+                type={GraphTypes.Line}
                 color={color}
-                datasets={results}
+                datasets={results as GraphDataset[]}
                 labels={(results[0] && results[0].labels) || []}
                 isInProgress={!filters.date_to}
-                dashboardItemId={dashboardItemId}
-                inSharedMode={inSharedMode}
+                dashboardItemId={dashboardItemId ? (String(dashboardItemId) as InsightShortId) : undefined}
```

Comment:
Is there a danger it'll convert a number to a string here... so the wrong type of data goes through, `String(number)` instead of `InsightShortId`?

Reply:
I've separated this out into two separate `insightShortId: InsightShortId` and `insightId: number` props